src/pycares/__init__.py

```diff
@@ -501,22 +501,32 @@ def gethostbyname(self, name, family, callback):
         _global_set.add(userdata)
         _lib.ares_gethostbyname(self._channel[0], parse_name(name), family, _lib._host_cb, userdata)
 
-    def query(self, name, query_type, callback):
-        self._do_query(_lib.ares_query, name, query_type, callback)
+    def query(self, name, query_type, callback, dnsclass=None):
+        self._do_query(_lib.ares_query, name, query_type, callback, dnsclass=dnsclass)
 
-    def search(self, name, query_type, callback):
-        self._do_query(_lib.ares_search, name, query_type, callback)
+    def search(self, name, query_type, callback, dnsclass=None):
+        self._do_query(_lib.ares_search, name, query_type, callback, dnsclass=dnsclass)
 
-    def _do_query(self, func, name, query_type, callback):
+    def _do_query(self, func, name, query_type, callback, dnsclass=None):
         if not callable(callback):
             raise TypeError('a callable is required')
 
         if query_type not in self.__qtypes__:
             raise ValueError('invalid query type specified')
 
+        DNSCLASS_MAP = {
+            "IN": _lib.C_IN,
+            "CH": _lib.C_CHAOS,
+            "HS": _lib.C_HS,
+            "NONE": _lib.C_NONE,
+            "ANY": _lib.C_ANY,
+        }
+
+        dnsclass_cares = DNSCLASS_MAP.get(dnsclass, _lib.C_IN)
```

Comment:
Should this mapping be elevated to the top level (akin to `QUERY_TYPE_x` symbols)?

Reply:
yeah

Reply:
Done.